src/System.Memory/src/System/SequencePosition.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+#pragma warning disable 1591
+
+namespace System
+{
+    public readonly struct SequencePosition : IEquatable<SequencePosition>
+    {
+        readonly object _segment;
+        readonly int _index;
+
+        public SequencePosition(object segment, int index)
+        {
+            _segment = segment;
+            _index = index;
+        }
+
+        public SequencePosition(object segment)
+        {
+            _segment = segment;
+            _index = 0;
+        }
+
+        public object Segment => _segment;
+
+        public int Index => _index;
+
+        public static bool operator ==(SequencePosition left, SequencePosition right) => left._index == right._index && left._segment == right._segment;
+        public static bool operator !=(SequencePosition left, SequencePosition right) => left._index != right._index || left._segment != right._segment;
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
```

Comment:
What is the rationale behind marking this as EditorBrowsableState.Never

Reply:
@KrzysztofCwalina

Reply:
I think we should remove the attributes from all but Equals(object obj).